app/presenters/publishing_api_presenters/statistics_announcement_redirect.rb

```diff
@@ -0,0 +1,52 @@
+require 'securerandom'
+
+module PublishingApiPresenters
+  class StatisticsAnnouncementRedirect
+    extend Forwardable
+    def_delegators :item, :base_path, :publication
+
+    attr_reader :content_id, :item
+
+    def initialize(item, _options = {})
+      @item = item
+      @content_id = SecureRandom.uuid
+    end
+
+    def content
+      {
+        base_path: base_path,
+        format: "redirect",
+        publishing_app: "whitehall",
+        redirects: [
+          {
+            path: base_path,
+            destination: Addressable::URI.parse(redirect_url).path,
+            type: "exact"
+          }
+        ]
+      }
+    end
+
+    def links
+      {}
+    end
+
+    def update_type
+      "major"
+    end
+
+  private
+
+    def redirect_url
+      publication_is_published? ? publication_url : item.redirect_url
+    end
+
+    def publication_is_published?
```

Comment:
In this case it is a little more _presenterish_, I agree. This wasn't too major a point to labour over.